### PR TITLE
feat(#9): storage init

### PR DESCRIPTION
**Pull request by h1alexbel**
@l3r8yJ take a look, please
ref #9

---

## PR-Codex overview
The focus of this PR is adding XML storage functionality with tests and updating dependencies.

### Detailed summary
- Added `anyhow`, `log`, `env_logger`, and `tempdir` dependencies
- Implemented XML storage functionality in `xml` module
- Added tests for XML storage creation with different file names

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by h1alexbel** [comment]
@l3r8yJ I suggest the following test convention: all unit test we specify in the same `.rs` file, while integration tests will go to the separate folder called `/tests`. WDYT?

File: src/xml/storage.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// SOFTWARE.
+use std::fs::File;
+
+use log::info;
+
+pub fn init() {
+    let path = "fakehub.xml";
+    info!("Initializing XML storage: {path}");
+    File::create(path).unwrap();
+    info!("{path} initialized.");
```
Comment (l3r8yJ) [inline src/xml/storage.rs:30]: let's wrap all paths in logs into `'` symbols

```suggestion
    info!("'{path}' initialized.");
```

File: src/xml/storage.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    fn clean() {
+        fs::remove_file("fakehub.xml").unwrap();
+    }
+
+    #[test]
+    fn creates_xml_storage() {
+        init();
+        let storage = "fakehub.xml";
+        let exists = Path::new(storage).exists();
+        assert!(exists, "storage file {storage} was not created, but should");
```
Comment (l3r8yJ) [inline src/xml/storage.rs:49]:
```suggestion
        assert!(exists, "storage file '{storage}' was not created, but should");
```

File: src/xml/storage.rs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+use std::fs::File;
+
+use log::info;
+
+pub fn init() {
+    let path = "fakehub.xml";
+    info!("Initializing XML storage: {path}");
+    File::create(path).unwrap();
```
[the remark this answers is not in the document]
Comment (h1alexbel) [reply src/xml/storage.rs:29]: @l3r8yJ you meant`.expect("some message after failure")`?

[the remark this answers is not in the document]
Comment (l3r8yJ) [reply src/xml/storage.rs:29]: @h1alexbel i meant smth like

```rust
let creating_file = File::create("storage.xml");
let storage = match creating_file {
    Ok(file) => file,
    Err(error) => panic!("Problem during creating the file: {:?}", error),
};
```

[the remark this answers is not in the document]
Comment (h1alexbel) [reply src/xml/storage.rs:29]: ah, got it, thanks

File: src/xml/storage.rs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+use std::fmt::Error;
+use std::fs::File;
+
+use log::info;
+
+pub fn touch_storage() {
+    let path = "fakehub.xml";
+    info!("Initializing XML storage: {path}");
+    let creating = File::create(path);
+    let _ = match creating {
```
Comment (l3r8yJ) [inline src/xml/storage.rs:31]: might be we should return the file from `touch_storage` and pass it to next function instead of sharing it by name?

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel last comment, take a look, please

File: src/xml/storage.rs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    use std::path::Path;
+
+    use crate::xml::storage::touch_storage;
+
+    fn clean() {
+        fs::remove_file("fakehub.xml").unwrap();
+    }
+
+    #[test]
+    fn creates_xml_storage() {
```
Comment (l3r8yJ) [inline src/xml/storage.rs:54]: best practise is to return `anyhow::Result<()>` and add `Ok(())` at the end of test

File: src/xml/storage.rs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+use std::fmt::Error;
+use std::fs::File;
+
+use log::info;
+
+pub fn touch_storage() -> Result<File, Error> {
```
Comment (l3r8yJ) [inline src/xml/storage.rs:27]: to make it testable i would suggest to pass name of file and using `Option`

```suggestion
pub fn touch_storage(name: Option<&str>) -> Result<File, Error> {
```
and use it like

```rust
name.unwrap_or("storage.xml");
```

File: src/xml/storage.rs [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+#[cfg(test)]
+mod tests {
+    use std::fs;
+    use std::path::Path;
+
+    use crate::xml::storage::touch_storage;
+
+    fn clean() {
+        fs::remove_file("fakehub.xml").unwrap();
```
[the remark this answers is not in the document]
Comment (h1alexbel) [reply src/xml/storage.rs:50]: @l3r8yJ we create this file inside `touch_storage()`, how we can create a temp file there? Or I miss something?

[the remark this answers is not in the document]
Comment (l3r8yJ) [reply src/xml/storage.rs]: @h1alexbel you can pass a temp directory and create file inside of it. I think that even better would be if `touch_storage` will accept directory or smth like that

**Comment by h1alexbel** [comment]
@l3r8yJ no, but I already applied the `temp dir` changes

**Review by l3r8yJ — APPROVED** [review]
@h1alexbel looks good to me! added some fix

**Comment by h1alexbel** [comment]
@l3r8yJ looks like `@rultor` is alive, let's try to mer with his help

**Comment by h1alexbel** [comment]
@rultor mereg

**Comment by h1alexbel** [comment]
`@rultor` is broken again

**Pull request merged**